### PR TITLE
Adding Scala language

### PR DESCRIPTION
**Pull request by techysubham**
### **Description**
Added Scala Language in ```lang``` folder

**Notes for Reviewers:**

**[Signed commits](https://github.com/devstrons/heloworld/blob/master/CONTRIBUTING.md)**

- [ ] Yes, I am a Member of **DEVSTRONS'**

**Discord ID** (e.g.: DEVSTRONS'#0238)
```AnonymousGeek#9515```

### CODE REVIEW

File: lang/scala.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Setting up Scala development environment
+
+- Step 1: Verifying Java Packages
+  
+  The first thing we need to have is a Java Software Development Kit(SDK) installed on the computer. We need to verify this SDK packages and if not installed then install them. Open the command window and type in the following commands:
+
+- For Windows
+```C:\Users\Your_PC_username>java -version```
+
+- Step 2: Now install Scala
```
Comment (Panquesito7) [inline lang/scala.md:39]:
```suggestion
- Step 2: Installing Scala
```

File: lang/scala.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- Step 2: Now install Scala 
+  
+  We are done with installing the java now let’s install the scala packages. The best option to download these packages is to download from the official site only: https://www.scala-lang.org/download/ 
+  The packages in the link above is the approximately of 100MB storage. Once the packages are downloaded then open the downloaded .msi file and install it.
+
+- Step 3:Testing and Running the Scala Commands
+  
+  Open the command prompt now and type in the following codes
+  ```C:\Users\Your_PC_username>scala```
+
```
Comment (Panquesito7) [inline lang/scala.md:48]:
```suggestion
We are done with installing the Java SDK. Now let’s install the Scala packages. The best option to download these packages is to download from the official site only: https://www.scala-lang.org/download/ 
The packages in the link above are approximately 100MB. Once the packages are downloaded, open the downloaded `.msi` file and install it.

- Step 3: Testing and running the Scala Commands
  
Open the command prompt now and type in the following codes:

  `C:\Users\Your_PC_username>scala`
```

File: lang/scala.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## Popular Scala IDEs/Editors
+
+- IntelliJ IDEA
+- Vim
+- Sublime Text
+- Atom Text editor
+- BlueJ
```
Comment (Panquesito7) [inline lang/scala.md:56]: Could you please link to their official websites (if applicable)?

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Great work! 🚀

**Comment by techysubham** [comment]
Made necessary changes , please check it.
Cheers...

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please add Wikipedia links when applicable to easily check the sources (like seen below).

**Comment by techysubham** [comment]
Added the Wikipedia links.
You can verify it.

File: lang/scala.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+- Scala uses an expressive typing system that ensures statistical abstraction is secure and consistent.
+- It’s easy to learn, especially for programmers with an object-oriented background with Java or similar language.
+- Scala is highly scalable, useful for building fault-tolerant, highly concurrent systems
+- It’s ideal for data analytics when supported by tools like Apache Spark
+
+
+## Setting up Scala development environment
+
+- Step 1: Verifying Java Packages
+- The first thing we need to have is a Java Software Development Kit(SDK) installed on the computer. We need to verify these SDK packages and if not installed then install them. Open the command window and type in the following commands:
```
Comment (Panquesito7) [inline lang/scala.md:33]: Provide a Wikipedia link to the Java SDK. Check other instances where it's needed as well.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Comment by techysubham** [comment]
Made the necessary changes.
You can verify it.

**Comment by aminoxix** [comment]
@techysubham can you pl add some emojis to your document of Scala? Refer to any other lang's docs pushed.

**Review by aminoxix — APPROVED** [review]
Great work - LGTM! 🚀

File: lang/scala.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+## 😎 Some cool projects in Scala 
+
+- [spark](https://github.com/apache/spark)
+- [lila](https://github.com/lichess-org/lila)
+- [SynapseML](https://github.com/microsoft/SynapseML)
+- [The Algorithms - Scala](https://github.com/TheAlgorithms/Scala)
+
+## 📑 Conclusion
+
+- Scala is a strong statically typed general-purpose programming language which supports both [object-oriented programming](https://en.wikipedia.org/wiki/Object-oriented_programming) and [functional programming](https://en.wikipedia.org/wiki/Functional_programming). Designed to be concise, many of Scala's design decisions are aimed to address criticisms of Java.
```
Comment (Panquesito7) [inline lang/scala.md:63]: Not sure if we should use a list here (same for other parts of the document).

**Review by Panquesito7 — COMMENTED** [review]
Otherwise, everything looks great! Thanks. 🚀

**Pull request merged**